msprime/species_trees.py

```diff
@@ -67,23 +108,55 @@ def parse_starbeast(tree, generation_time, branch_length_units="myr"):
 
 def parse_species_tree(tree, Ne, branch_length_units="gen", generation_time=None):
     """
-    Method to parse species trees in Newick
-    (https://en.wikipedia.org/wiki/Newick_format) format.
-
-    Trees are assumed to be rooted and ultrametric and branch lengths
-    must be included and correspond to time, either in units of millions
-    of years ("myr"), years ("yr"), or generations ("gen"; default).
-    Leafs must be named. An example for an accepted tree string in
-    Newick format is:
-    (((human:5.6,chimpanzee:5.6):3.0,gorilla:8.6):9.4,orangutan:18.0)
-    The newick library (https://pypi.org/project/newick/) is used for
-    tree parsing. The tree string can end with a semi-colon, but this is
-    not required.
-
-    - An estimate of the effective population size Ne should be
-        specified.
-    - If and only if the branch lengths are not in units of
-        generations, the generation time should be specified.
+    This function parses a species tree in
+    `Newick <https://en.wikipedia.org/wiki/Newick_format>`_ format and defines a
+    simulation model according to the species tree. The tree is assumed to be
+    rooted and ultrametric and branch lengths must be included and correspond to
+    time, either in units of millions of years, years, or generations. Leafs must
+    be named.
+
+    After reading the input tree, this function defines a population configuration
+    for each terminal node in the tree, corresponding to extant species. These
+    population configurations store the species' name and population size. The
+    specified Ne is used as the size of all populations. Additionally, one or more
+    mass migration events are defined for each internal node, with the time of the
+    mass migration set according to the age of the node in the species tree. `n`-1
```

Comment:
using :math: can be a bit nicer for this, i.e., ``:math:`n - 1` mass migration...``